lightning/src/util/events.rs

```diff
@@ -382,6 +382,41 @@ pub enum Event {
 #[cfg(test)]
 		error_data: Option<Vec<u8>>,
 	},
+	/// Indicates that a probe payment we sent returned successful, i.e., only failed at the destination.
+	ProbePaymentSuccessful {
+		/// The id returned by [`ChannelManager::send_probe_payment`].
+		///
+		/// [`ChannelManager::send_probe_payment`]: crate::ln::channelmanager::ChannelManager::send_probe_payment
+		payment_id: PaymentId,
+		/// The hash generated by [`ChannelManager::send_probe_payment`].
+		///
+		/// [`ChannelManager::send_probe_payment`]: crate::ln::channelmanager::ChannelManager::send_probe_payment
+		payment_hash: PaymentHash,
+		/// The payment path that was successful.
+		path: Vec<RouteHop>,
+	},
+	/// Indicates that a probe payment we sent failed at an intermediary node on the path.
+	ProbePaymentFailed {
+		/// The id returned by [`ChannelManager::send_probe_payment`].
+		///
+		/// [`ChannelManager::send_probe_payment`]: crate::ln::channelmanager::ChannelManager::send_probe_payment
+		payment_id: PaymentId,
+		/// The hash generated by [`ChannelManager::send_probe_payment`].
+		///
+		/// [`ChannelManager::send_probe_payment`]: crate::ln::channelmanager::ChannelManager::send_probe_payment
+		payment_hash: PaymentHash,
+		/// The payment path that failed.
+		path: Vec<RouteHop>,
+		/// The channel responsible for the failed probe.
+		///
+		/// Note that for route hints or for the first hop in a path this may be an SCID alias and
+		/// may not refer to a channel in the public network graph. These aliases may also collide
+		/// with channels in the public network graph.
+		///
+		/// If this is `Some`, then the corresponding channel should be avoided when the payment is
+		/// retried. May be `None` for older [`Event`] serializations.
+		short_channel_id: Option<u64>,
```

Comment:
Shouldn't need to be an `Option` since there are no older versions having this event, unlike `PaymentPathFailed`.

Reply:
The reason why I originally went with an `Option` here is that I'm not sure if the `short_channel_id: Option<u64>` returned by `onion_utils::process_onion_failure` (cf. [here](https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/ln/channelmanager.rs#L3841)) will really really always be `Some(..)` in this case. I now put an `unwrap()` there with https://github.com/lightningdevkit/rust-lightning/pull/1567/commits/af969a5d30c87c0477e75e44dc03e21de3e5b49d, but I'm currently not entirely positive this is safe. The alternative would be to return a default value in the case we have a `None`, e.g., `u64::max_value()` again, however, this is essentially a poor man's `Option` then?

Reply:
Ah, right. Good eye. Probably should remove the older serialization language from the docs, though. Maybe for `PaymentPathFailed` as well?

Reply:
Good idea, removed the corresponding lines from the docs now.